spring-web/src/main/java/org/springframework/http/server/reactive/ReactorServerHttpResponse.java

```diff
@@ -56,7 +56,6 @@ class ReactorServerHttpResponse extends AbstractServerHttpResponse implements Ze
 
 	public ReactorServerHttpResponse(HttpServerResponse response, DataBufferFactory bufferFactory) {
 		super(bufferFactory, new HttpHeaders(new NettyHeadersAdapter(response.responseHeaders())));
```

Comment:
```suggestion
		super(bufferFactory, new HttpHeaders(new NettyHeadersAdapter(Objects.requireNonNull(response, "HttpServerResponse must not be null").responseHeaders())));
```

Although it will throw a `NullPointerException` instead of an `IllegalArgumentException`, if we use `Objects.requireNonNull` we can still have a custom error message.

So let's go with that.

Reply:
Made the change